system/version/goos/activate_tok_darwin.go

```diff
@@ -0,0 +1,41 @@
+package goos
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ironstar-io/tokaido/constants"
+	"github.com/ironstar-io/tokaido/system/fs"
+	"github.com/ironstar-io/tokaido/utils"
+)
+
+// ActivateSavedVersion - Copies the specified version (which may be downloaded previously) into /usr/local/bin on macOS
+func ActivateSavedVersion(version string) bool {
+	// Check that the version is downloaded already
+	p := filepath.Join(fs.HomeDir(), constants.BaseInstallPathDarwin, version, "tok")
+	if fs.CheckExists(p) != true {
+		utils.DebugString("Tokaido version [" + version + "] was not found in ~/.tok/bin, downloading a new copy...")
+
+		_, err := DownloadTokBinary(version)
+		if err != nil {
+			fmt.Println("Unexpected error downloading that version: " + err.Error())
+			os.Exit(1)
+		}
+	}
+
+	// Remove any existing copy of Tokaido at /usr/local/bin/tok
+	fs.Remove(constants.ActiveBinaryPathDarwin)
```

Comment:
Does this still work if the binary being removed is the binary that's running? That might've been a problem that symlinking solved?

Reply:
Only if the next parts fail. This binary is in at the time it's being removed, so removing it isn't a problem. 

You can test this (and I did, just to be sure) by running something that keeps tok running, like `tok config` and then removing tokaido from your system, you'll see that you can keep using and even saving config in `tok config` even though it's binary is deleted from disk.

Reply:
Not sure about Windows though, I feel like I've run into that before

Reply:
We're going to find out real soon :)